src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/DiagnosticDescriptorExtensions.cs

```diff
@@ -55,6 +56,41 @@ public static ImmutableArray<string> ImmutableCustomTags(this DiagnosticDescript
             return effectiveSeverity;
         }
 
+        /// <summary>
+        /// Gets document-level effective severity of the given <paramref name="descriptor"/> accounting for severity configurations from both the following sources:
+        /// 1. Compilation options from ruleset file, if any, and command line options such as /nowarn, /warnaserror, etc.
+        /// 2. Analyzer config documents at the document root directory or in ancestor directories.
+        /// </summary>
+        public static ReportDiagnostic GetEffectiveSeverity(this DiagnosticDescriptor descriptor, CompilationOptions compilationOptions, SyntaxTree tree, AnalyzerOptions analyzerOptions)
+        {
+            var effectiveSeverity = descriptor.GetEffectiveSeverity(compilationOptions);
+
+            // Apply analyzer config options, unless configured with a non-default value in compilation options.
+            // Note that compilation options (/nowarn, /warnaserror) override analyzer config options.
+            if (!compilationOptions.SpecificDiagnosticOptions.TryGetValue(descriptor.Id, out var reportDiagnostic) ||
+                reportDiagnostic == ReportDiagnostic.Default)
+            {
+                // First check for tree-level analyzer config options.
+                var analyzerConfigOptions = analyzerOptions.AnalyzerConfigOptionsProvider.GetOptions(tree);
```

Comment:
Seems like the options returned here only contain [StructuredAnalyzerConfigOptions](https://sourceroslyn.io/Microsoft.CodeAnalysis.Workspaces/StructuredAnalyzerConfigOptions.cs.html#9b9b8e407bd0092f), which excludes all diagnostic severity configuration entries in editorconfig. This seems to have happened with recent options related refactorings, which would have also broken the existing `GetEffectiveSeverity` extension methods in this file. I am going to create a separate PR to fix this regression.

Reply:
Opened https://github.com/dotnet/roslyn/pull/66497 for this.